src/frequenz/core/enum.py

```diff
@@ -48,8 +55,24 @@ def __init__(self, value: Any, message: str) -> None:
         self.message = message
 
 
+def deprecated_member(value: ValueT, message: str) -> ValueT:
+    """Mark an enum member as deprecated.
+
+    Please read the [`Enum`][frequenz.core.enum.Enum] documentation for details and
+    examples.
+
+    Args:
+        value: The value of the deprecated enum member.
```

Comment:
I'd make a difference between the doc for the argument and the return value.
Here I'd suggest something around:
```
value: The value of the enum member to mark as deprecated.
```